.github/workflows/docgen-check.yml

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A workflow to check if generated docs are up to date
+name: doc-gen check
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  docgen-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2 # refs/pull/:prNumber/merge
+
+      - name: Setup Java and Maven Env
+        uses: actions/setup-java@v3
+        with:
+          distribution: adopt
+          java-version: 8
+
+      - name: build deps
+        run: mvn install -pl 'aggregator' -am -DskipTests
+
+      - name: doc-gen
+        run: mvn antrun:run@update_config_docs -pl dist
+
+      - name: check if existing modified files
+        run: mvn exec:exec@if_modified_files -P pre-merge -pl dist
```

Comment:
these three invocations of maven should be just one maven step to the tune of 
```
 mvn verify -pl dist -am -DskipTests -Dskip -Dmaven.javadoc.skip
 ```
 
 that takes in 2+ minutes. It will take care of style checks, default shim version and conf generation.

Reply:
sounds good, I will refactor the check to cover more

Reply:
I was intentional to separate stages, so people could quickly figure out which stage their job failed. But as they still need to check the log, leverage a single command to cover should be fine